README.md

```diff
@@ -107,54 +110,79 @@ keys_gen.sh can be used to generate random passwords for each service.env.
 
 ### Generating random passwords using keys_gen.sh
 
-Running keys_gen.sh will generate random passwords for the all http servers.
+1. Generating demo certificate authority KeyPair and certificate
 
-```
-$ cd <fdo-pri-src>/component-samples/demo/scripts
-$ ./keys_gen.sh
-```
+    ```shell
+      bash demo-ca.sh
+    ```
 
-A message "Key generation completed." will be displayed on the console.
+    **NOTE**: Configure the properties of `demo-CA` by updating `root-ca.conf`.
 
-Credentials for each service will be generated the creds directory `<fdo-pri-src>/component-sample/demo/`.
+2. Generating Server and Client Keypair and certificates.
 
-```
-creds/aio/service.env
-creds/manufacturer/service.env
-creds/owner/service.env
-creds/reseller/service.env
-creds/rv/service.env
-```
+    ```shell
+    bash web_csr_req.sh
+    bash user_csr_req.sh
+    ```
 
-Replace the service.env in the demo folders with the generated ones in the creds folder.
+    **NOTE**: Both Server and Client certificate are signed by the previous generated demo-CA. Moreover, we can configure the properties of Server and Client certificate by updating `web-server.conf` and `client.conf` respectively.
+
+
+3. Running keys_gen.sh will generate random passwords for the all http servers and creates `secrets` folder containing all the required `.pem` files of Client, CA and Server component.
+
+    ```
+    $ cd <fdo-pri-src>/component-samples/demo/scripts
+    $ ./keys_gen.sh
+    ```
+
+    A message "Key generation completed." will be displayed on the console.
+
+    Credentials will be stored in the `secrets` directory within `<fdo-pri-src>/component-sample/demo/scripts`.
+
+4. Copy both `secrets/` and `service.env` file from  `<fdo-pri-src>/component-sample/demo/scripts`  folder to the individual components.
+
+    ```
+    $ cd <fdo-pri-src>/component-samples/demo/scripts
+    $ cp -r ./secrets/. ../replace-with-FDO-component
+    $ cp service.env ../replace-with-FDO-component
+    ```
+
+    **NOTE:** Don't replace `service.env` present in the database component with generated `service.env` in `scripts` folder.
 
-```
-$ cd <fdo-pri-src>/component-samples/demo/scripts
-$ cp -r ./creds/. ../
-```
 
 ### Specifying Subject alternate names for the Web/HTTPS self-signed certificate.
 
 When the http server starts for the first time it will generate a self-signed certificate for https protocol.
 
-The subject name of the self-signed certificate is defined in the service.yml under 
+The subject name of the self-signed certificate is defined in the  `web-server.conf` under
 ```
-http-server:
-    subject_names:
-    - DNS:host.docker.internal
-    - IP:127.0.0.1
+#[ req_ext ]
+  subjectAltName = @alt_names
+
+[ alt_names ]
+  #Replace or add new DNS and IP entries with the ones required by the HTTPs/Web Service.
+  DNS.1 = www.example.com
+  DNS.2 = test.example.com
+  DNS.3 = mail.example.com
+  DNS.4 = www.example.net
+  IP.1 = 192.0.0.0
```

Comment:
can we give something other than 192? just a suggestion

Reply:
Done.